File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.js
function splitString(myStr) {
    let result = [];
    let startIndex = 0;
    const reg = new RegExp('[abc]');

    for (let i = 0; i <= myStr.length; i++) {
        if (reg.test(myStr[i]) || i === myStr.length) {
            if (startIndex !== i) result.push(myStr.slice(startIndex, i));
            startIndex = i + 1;
        }
    }
    
    if (result.length === 0) result.push("EMPTY");
    
    return result;
}

function solution(myStr) {
    let answer = [];
    
    answer = splitString(myStr);
    
    return answer;
}

File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.java
import java.util.*;

class Solution {
    public String[] solution(String myStr) {
        String[] answer = {};
        answer = myStr.split("[a-c]+");

        List<String> list = new ArrayList<>();
        for(int i = 0; i< answer.length; i++){
            if(!"".equals(answer[i])) list.add(answer[i]);
        }

        if(list.size() == 0) list.add("EMPTY");

        answer = new String[list.size()];
        for(int i = 0; i < list.size(); i++) answer[i] = list.get(i);

        return answer;
    }
}